src/main/resources/schema/all_merged_subjects_v3_3_final.json

```diff
@@ -174,24 +174,6 @@
                             "alias": "total",
                             "type": "TOTAL"
                         }
-                    ],
-                    "foreignKeys": [
-                        {
-                            "fields": [
-                                "case_id"
-                            ],
-                            "type": "SINGLE",
-                            "tableName": "somatic_mutation_hg38_gdc_r36_v3_3_final",
-                            "tableAlias": "mutations"
-                        },
-                        {
-                            "fields": [
-                                "ResearchSubjects"
```

Comment:
If I understand correctly, this gets removed because files will naturally link without a foreign key.

Reply:
Dion, I think you removed too much from the foreignKeys block. I think the ResearchSubjects section should not have been removed. (At least I don't think this is related to removing the incorrect relationships to the mutations table)

Reply:
If I was following Rich the other day, the graph has enough definition to find the relationships needed for ResearchSubjects.

Dion was able to write a test to prove this and pushed it to the [cda-python branch](https://github.com/CancerDataAggregator/cda-python/pull/202/files/9612395573915afdb6ce56ec80bd3fd314688839..5efd7acd665bae7169efb4f522e6442e180ba4eb) he was working on.  Not only are the counts the same in the results, but the resulting SQL has the same operations.

# integration with the graph changes
`
            Query:SELECT results.* EXCEPT(rn) FROM (SELECT ROW_NUMBER() OVER (PARTITION BY File.id, Subject.id, _ResearchSubject.id, _Specimen.id) as rn,
File.id AS file_id, File.identifier AS file_identifier, File.label AS label, File.data_category AS data_category, File.data_type AS data_type,
File.file_format AS file_format, File.associated_project AS file_associated_project, File.drs_uri AS drs_uri, File.byte_size AS byte_size, File.checksum
AS checksum, File.data_modality AS data_modality, File.imaging_modality AS imaging_modality, File.dbgap_accession_number AS dbgap_accession_number,
File.imaging_series AS imaging_series, Subject.id as subject_id, _ResearchSubject.id as researchsubject_id, _Specimen.id as specimen_id FROM
gdc-bq-sample.dev.all_merged_files_v3_3_final AS File LEFT JOIN UNNEST(File.Subjects) AS _Subjects LEFT JOIN UNNEST(File.ResearchSubjects) AS
_ResearchSubjects LEFT JOIN UNNEST(File.Specimens) AS _Specimens LEFT JOIN gdc-bq-sample.dev.all_merged_subjects_v3_3_final AS Subject ON _Subjects =
Subject.id LEFT JOIN gdc-bq-sample.dev.somatic_mutation_hg38_gdc_r36_v3_3_final AS Mutation ON Subject.id = Mutation.cda_subject_id LEFT JOIN
UNNEST(Subject.ResearchSubject) AS _ResearchSubject ON _ResearchSubject.id = _ResearchSubjects LEFT JOIN UNNEST(_ResearchSubject.Specimen) AS _Specimen
ON _Specimen.id = _Specimens WHERE ((IFNULL(UPPER(File.file_format), ‘’) = UPPER(‘tsv’)) AND (IFNULL(UPPER(Mutation.SYMBOL), ‘’) LIKE UPPER(‘TP53%‘))))
as results WHERE rn = 1
            Offset: 0
            Count: 100
            Total Row Count: 119383
            More pages: True
            `
# prod with out the graph changes
`
            Query:SELECT results.* EXCEPT(rn) FROM (SELECT ROW_NUMBER() OVER (PARTITION BY File.id, Subject.id, _ResearchSubject.id, _Specimen.id) as rn,
File.id AS file_id, File.identifier AS file_identifier, File.label AS label, File.data_category AS data_category, File.data_type AS data_type,
File.file_format AS file_format, File.associated_project AS file_associated_project, File.drs_uri AS drs_uri, File.byte_size AS byte_size, File.checksum
AS checksum, File.data_modality AS data_modality, File.imaging_modality AS imaging_modality, File.dbgap_accession_number AS dbgap_accession_number,
File.imaging_series AS imaging_series, Subject.id as subject_id, _ResearchSubject.id as researchsubject_id, _Specimen.id as specimen_id FROM
broad-dsde-prod.cda_prod.all_merged_files_v3_3_final AS File LEFT JOIN UNNEST(File.Subjects) AS _Subjects LEFT JOIN UNNEST(File.ResearchSubjects) AS
_ResearchSubjects LEFT JOIN UNNEST(File.Specimens) AS _Specimens LEFT JOIN broad-dsde-prod.cda_prod.all_merged_subjects_v3_3_final AS Subject ON
_Subjects = Subject.id LEFT JOIN broad-dsde-prod.cda_prod.somatic_mutation_hg38_gdc_r36_v3_3_final AS Mutation ON Subject.id = Mutation.cda_subject_id
LEFT JOIN UNNEST(Subject.ResearchSubject) AS _ResearchSubject ON _ResearchSubject.id = _ResearchSubjects LEFT JOIN UNNEST(_ResearchSubject.Specimen) AS
_Specimen ON _Specimen.id = _Specimens WHERE ((IFNULL(UPPER(File.file_format), ‘’) = UPPER(‘tsv’)) AND (IFNULL(UPPER(Mutation.SYMBOL), ‘’) LIKE
UPPER(‘TP53%’)))) as results WHERE rn = 1
            Offset: 0
            Count: 100
            Total Row Count: 119383
            More pages: True`

@ahaessly - Was this the case you were concerned with or did you have a different one?

Reply:
I guess I'm wondering why we decided to remove the foreign key from ResearchSubject.id -> files.ResearchSubjects. This relationship is not invalid. I thought we were just removing the 2 invalid relationships to the mutations table. In general, I'm not opposed to removing the ResearchSubject.id -> files.ResearchSubjects if we find that it is redundant (because we have a reciprocal relationship from files -> research subjects). But then I think we should also remove the other redundant relationships like Subject.id -> files.Subjects and Specimen.id -> files.Specimens. It just didn't make sense to me why we removed that one when it had nothing to do with the invalid relationships to the mutations tables. It just seems like a risky change that is unrelated to the fix we are making in this PR.

Reply:
(the conversation moved to Slack where we've asked our teammate to test)